lib/common/nvpair.c

```diff
@@ -463,14 +479,15 @@ pcmk__cmp_nvpair_blocks(gconstpointer a, gconstpointer b, gpointer user_data)
     const gint a_is_higher = unpack_data->overwrite? 1 : -1;
     const gint b_is_higher = -a_is_higher;
 
-    if (a == NULL && b == NULL) {
-        return 0;
-    } else if (a == NULL) {
-        return b_is_higher;
+    // NULL values have lowest priority
+    if (a == NULL) {
+        return (b == NULL)? 0 : b_is_higher;
+
     } else if (b == NULL) {
         return a_is_higher;
     }
 
+    // A particular XML ID can be specified as having highest priority
```

Comment:
// XML ID is unique with all the blocks so we can't  have them both equal first ID
// and checking scores isn't needed here either

Reply:
Thanks, I'll update the comment